src/app/common/context/AppContext.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { INotification } from '@app/common/components/Notifications';
+
+interface IAppContext {
+  pushNotification: (notification: INotification) => void;
+  dismissNotification: (key: string) => void;
+  notifications: INotification[];
+}
+
+const appContextDefaultValue = {} as IAppContext;
+export const AppContext = React.createContext<IAppContext>(appContextDefaultValue);
+
+interface IAppContextProvider {
+  children: React.ReactNode;
+}
+
+export const AppContextProvider: React.FunctionComponent<IAppContextProvider> = ({
```

Comment:
Okay, I know I've said this before and backed down, but part of me is worried that if we introduce an `AppContext`, in the future we'll end up shoving a bunch of unrelated stuff together in there any time we have a need for something to be globally available. Maybe that's fine, but I feel like it could become a mess. I bet you if we'd called it `AppContext` when we started out with `NetworkContext`, we'd have ended up lumping in the stuff that is now in `LocalStorageContext` and `PollingContext` etc.

It's not too late to call this `NotificationContext` :) but also I know I'm being paranoid and it's fine.

Reply:
I like NotificationContext :)

You convinced me at the point we created AppLayoutContextProvider 😆

Reply:
Ha, fair. That one is maybe a bit rough lol